mhkit/dolfyn/io/rdi.py

```diff
@@ -1095,36 +1098,36 @@ def read_vmdas(self):
 
         # 1st lat/lon position after previous ADCP ping
         # This byte is in hundredths of seconds (10s of milliseconds):
-        utc_time_first_fix = tmlib.timedelta(milliseconds=(int(fd.read_ui32(1) / 10)))
+        utc_time_first_fix = tmlib.timedelta(milliseconds=(int(fd.read_ui32(1) * 0.1)))
         ens.clock_offset_UTC_gps[k] = (
-            fd.read_i32(1) / 1000
+            fd.read_i32(1) * 0.001
         )  # "PC clock offset from UTC" in ms
-        latitude_first_gps = fd.read_i32(1) * self._cfac
-        longitude_first_gps = fd.read_i32(1) * self._cfac
+        latitude_first_gps = fd.read_i32(1) * self._cfac32
+        longitude_first_gps = fd.read_i32(1) * self._cfac32
 
         # Last lat/lon position prior to current ADCP ping
-        utc_time_fix = tmlib.timedelta(milliseconds=(int(fd.read_ui32(1) / 10)))
+        utc_time_fix = tmlib.timedelta(milliseconds=(int(fd.read_ui32(1) * 0.1)))
         ens.time_gps[k] = tmlib.date2epoch(date_utc + utc_time_fix)[0]
-        ens.latitude_gps[k] = fd.read_i32(1) * self._cfac
-        ens.longitude_gps[k] = fd.read_i32(1) * self._cfac
+        ens.latitude_gps[k] = fd.read_i32(1) * self._cfac32
+        ens.longitude_gps[k] = fd.read_i32(1) * self._cfac32
 
-        ens.avg_speed_gps[k] = fd.read_ui16(1) / 1000
-        ens.avg_dir_gps[k] = fd.read_ui16(1) * 180 / 2**15  # avg true track
+        ens.avg_speed_gps[k] = fd.read_ui16(1) * 0.001
+        ens.avg_dir_gps[k] = fd.read_ui16(1) * self._cfac16  # avg true track
         fd.seek(2, 1)  # avg magnetic track
-        ens.speed_made_good_gps[k] = fd.read_ui16(1) / 1000
-        ens.dir_made_good_gps[k] = fd.read_ui16(1) * 180 / 2**15
+        ens.speed_made_good_gps[k] = fd.read_ui16(1) * 0.001
+        ens.dir_made_good_gps[k] = fd.read_ui16(1) * self._cfac16
         fd.seek(2, 1)  # reserved
         ens.flags_gps[k] = int(np.binary_repr(fd.read_ui16(1)))
         fd.seek(6, 1)  # reserved, ADCP ensemble #
 
         # ADCP date time
         utim = fd.read_ui8(4)
         date_adcp = tmlib.datetime(utim[0] + utim[1] * 256, utim[3], utim[2])
-        time_adcp = tmlib.timedelta(milliseconds=(int(fd.read_ui32(1) / 10)))
+        time_adcp = tmlib.timedelta(milliseconds=(int(fd.read_ui32(1) * 0.1)))
```

Comment:
What the purpose of this change from division to multiplication?

Reply:
Copying @jmcvey3 response from our meeting - the purpose of these changes is to be consistent with the rest of DOLfYN's implementation